src/Data/Fin/Subset/Properties.agda

```diff
@@ -41,9 +42,16 @@ open import Relation.Unary using (Pred; Decidable; Satisfiable)
 drop-there : ∀ {s n x} {p : Subset n} → suc x ∈ s ∷ p → x ∈ p
 drop-there (there x∈p) = x∈p
 
+drop-not-there : ∀ {s n x} {p : Subset n} → suc x ∉ s ∷ p → x ∉ p
+drop-not-there x∉sp x∈p = contradiction (there x∈p) x∉sp
+
 drop-∷-⊆ : ∀ {n s₁ s₂} {p₁ p₂ : Subset n} → s₁ ∷ p₁ ⊆ s₂ ∷ p₂ → p₁ ⊆ p₂
 drop-∷-⊆ p₁s₁⊆p₂s₂ x∈p₁ = drop-there (p₁s₁⊆p₂s₂ (there x∈p₁))
 
+drop-∷-⊂ : ∀ {n s} {p₁ p₂ : Subset n} → s ∷ p₁ ⊂ s ∷ p₂ → p₁ ⊂ p₂
+drop-∷-⊂ {s = inside} (_ , zero , x∈sp₂ , x∉sp₁) = contradiction here x∉sp₁
```

Comment:
Since you're case-splitting on the argument (the 2nd part of the 4-tuple, it seems), and don't use `s` at all, why bother defining/refining it?

Reply:
AGDA does not typecheck otherwise.

Reply:
I see (and I think I get it).